src/thrift-netcore/Transport/TcpClientExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Thrift.Transport
+{
+    public static class TcpClientExtensions
+    {
+        public static async Task ClientConnectAsync(this TcpClient tcpClient, string host, int port)
+        {
+#if NETSTANDARD1_4 || NETSTANDARD1_5
+            IPAddress ip;
+            if (IPAddress.TryParse(host, out ip)) // if host is ip, parse it and connect
+            {
+                await tcpClient.ConnectAsync(ip, port);
+            }
+            else
+            {
+                IPAddress[] addresses = await Dns.GetHostAddressesAsync(host);// not ip, resolve it and get IP(s)
+                
+                if (addresses.Length > 0)
+                {
```

Comment:
Nitpicking here but could you use DateTime.UtcNow instead of DateTime.Now it is ~100 times faster (see https://blogs.msdn.microsoft.com/kirillosenkov/2012/01/10/datetime-utcnow-is-generally-preferable-to-datetime-now/)

Reply:
So good! I didn't realize that before